src/main/java/com/gluonhq/substrate/util/windows/MSIBundler.java

```diff
@@ -0,0 +1,327 @@
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Writer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.text.MessageFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.gluonhq.substrate.model.ReleaseConfiguration.DEFAULT_APP_VERSION;
+
+public class MSIBundler {
+
+    private final ProcessPaths paths;
+    private final InternalProjectConfiguration projectConfiguration;
+    private final String sourceOS;
+    private final Path rootPath;
+
+    public MSIBundler(ProcessPaths paths, InternalProjectConfiguration projectConfiguration) {
+        this.paths = paths;
+        this.projectConfiguration = projectConfiguration;
+        this.sourceOS = projectConfiguration.getTargetTriplet().getOs();
+        this.rootPath = paths.getSourcePath().resolve(sourceOS);
+    }
+
+    public boolean createPackage(boolean sign) throws IOException, InterruptedException {
+
+        if (WixTool.CANDLE.getPath() == null || WixTool.LIGHT.getPath() == null) {
+            Logger.logSevere("Please make sure to install Wix Toolset (https://wixtoolset.org/) before proceeding.");
+            return false;
+        }
+
+        final String appName = projectConfiguration.getAppName();
+        final String version = Optional.ofNullable(projectConfiguration.getReleaseConfiguration().getVersionName()).orElse(DEFAULT_APP_VERSION);
+        Path localAppPath = paths.getAppPath().resolve(appName + ".exe");
+        if (!Files.exists(localAppPath)) {
+            throw new IOException("Error: " + appName + ".exe not found");
+        }
+        Logger.logInfo("Building exe for " + localAppPath);
+
+        /**
+         * Create directories and copy resources
+         */
+        Path tmpMSI = paths.getTmpPath().resolve("tmpMSI");
+        if (Files.exists(tmpMSI)) {
+            FileOps.deleteDirectory(tmpMSI);
+        }
+        Files.createDirectories(tmpMSI);
+
+        Path userAssets = rootPath.resolve(Constants.WIN_ASSETS_FOLDER);
+        // copy assets to gensrc/windows
+        Path windowsGenSrcPath = paths.getGenPath().resolve(sourceOS);
+        Path windowsAssetPath = windowsGenSrcPath.resolve(Constants.WIN_ASSETS_FOLDER);
+        Files.createDirectories(windowsAssetPath);
+        // Copy system resources and over-write it with user provided resources
+        FileOps.copyDirectoryFromResources("/native/windows/assets", windowsAssetPath);
+        Logger.logInfo("Default icon.ico image generated in " + windowsAssetPath + ".\n" +
+                "Consider copying it to " + rootPath + " before performing any modification");
+        if (Files.exists(userAssets)) {
+            FileOps.copyDirectory(userAssets, windowsAssetPath);
```

Comment:
This works, but this is not how we do it for the resources and assets of other platforms.
We don't copy user assets into gensrc, which is only for default resources. 

In the next step you could simply do:
```
if (Files.exists(userAssets)) {
    FileOps.copyDirectory(userAssets, config);
} else {
    FileOps.copyDirectory(windowsAssetPath, config);
}
```